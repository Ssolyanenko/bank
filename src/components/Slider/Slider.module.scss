.slider {
  white-space: nowrap;
  transition: ease 1s;
  position: absolute;
  width: 100%;
  filter: brightness(0.35);
}
