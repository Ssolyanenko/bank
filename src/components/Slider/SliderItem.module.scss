.sliderItem {
  display: inline-block;
  width: 100%;
  overflow: hidden;

  img {
    width: 100%;
    height: 100vh;
    min-height: 726px;
    background-repeat: no-repeat;
    overflow: hidden;
    background-size: cover;
    background-position: center;
  }
}
