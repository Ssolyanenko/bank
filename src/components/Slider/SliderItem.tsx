import React, { FC, ReactElement } from 'react';

import classes from './SliderItem.module.scss';

interface Props {
  slide: string;
}

export const SliderItem: FC<Props> = ({ slide }): ReactElement => (
  <div className={classes.sliderItem}>
    <img src={slide} alt={slide} />
  </div>
);
