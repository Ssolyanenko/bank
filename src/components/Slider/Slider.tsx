import React, { FC, ReactElement } from 'react';

import { SliderItem } from './SliderItem';
import classes from './Slider.module.scss';

interface Props {
  slides: string[];
  activeSlide: number;
}

export const Slider: FC<Props> = ({ slides, activeSlide }): ReactElement => (
  <div className={classes.slider} style={{ transform: `translateX(${-100 * activeSlide}%)` }}>
    {slides.map((slide, index) => (
      <SliderItem slide={slide} key={slide + index} />
    ))}
  </div>
);
