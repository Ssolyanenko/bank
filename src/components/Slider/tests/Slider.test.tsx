import React from 'react';
import { shallow } from 'enzyme';

import { Slider } from 'components/Slider/Slider';

describe('<Slider />', (): void => {
  test('should make a snapshot', (): void => {
    const props = {
      slides: [
        'images/landingBack.png',
        'images/landingBack2.png',
        'images/landingBack3.png',
        'images/landingBack4.png',
      ],
      activeSlide: 0,
    };
    const wrapper = shallow(<Slider {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });
});
