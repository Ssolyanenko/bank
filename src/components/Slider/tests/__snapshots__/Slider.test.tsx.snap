// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should make a snapshot 1`] = `
<div
  className="slider"
  style={
    {
      "transform": "translateX(0%)",
    }
  }
>
  <SliderItem
    slide="images/landingBack.png"
  />
  <SliderItem
    slide="images/landingBack2.png"
  />
  <SliderItem
    slide="images/landingBack3.png"
  />
  <SliderItem
    slide="images/landingBack4.png"
  />
</div>
`;
